201-encrypt-create-new-vm-gallery-image/azuredeploy.json

```diff
@@ -52,31 +52,42 @@
                 "description": "Name of Subnet to which the VM NIC belongs to"
             }
         },
-        "AADClientID": {
+        "aadClientID": {
             "type": "string",
             "metadata": {
                 "description": "Client ID of AAD app which has permissions to KeyVault"
             }
         },
-        "AADClientSecret": {
+        "aadClientSecret": {
             "type": "securestring",
             "metadata": {
                 "description": "Client Secret of AAD app which has permissions to KeyVault"
             }
         },
-        "KeyVaultResourceID": {
+        "keyVaultResourceID": {
             "type": "string",
             "metadata": {
                 "description": "ResourceID of the KeyVault to place the volume encryption key"
             }
         },
-        "KeyVaultURL": {
+        "keyVaultURL": {
             "type": "string",
             "metadata": {
                 "description": "URL of the KeyVault to place the volume encryption key"
             }
         },
-        "KeyEncryptionKeyURL": {
+        "useExistingKek": {
+            "type": "string",
+            "defaultValue": "nokek",
+            "allowedValues": [
+                "nokek",
+                "kek"
+            ],
+            "metadata": {
+                "description": "Select kek if the secret should be encrypted with a key encryption key and pass explicit keyEncryptionKeyURL. For nokek, you can keep keyEncryptionKeyURL empty."
+            }
+        },
+        "keyEncryptionKeyURL": {
             "type": "string",
```

Comment:
If you had the name of the key vault, resource group of key vault and secret name, can't you get it like below?

"secretUrl": "[reference(resourceId(resourceGroup().name, 'Microsoft.KeyVault/vaults/secrets', 'vaultname', 'secretname'), '2015-06-01').secretUriWithVersion]"